lib/services/pdf_renderer/layout.js

```diff
@@ -30,13 +30,17 @@ export function autoLayoutItems(doc, componentSizes, printSettings) {
       h: (pageSize.h - totalVerticalMargin)
     }
 
+  // We sort the componentSizes by size, to make it easier to reuse the same page,
+  // even for different components, when using the componentMerging setting
+  componentSizes = [...componentSizes].sort((a, b) => (a.size.w - b.size.w) - (a.size.h - b.size.h))
```

Comment:
I like the pragmatic use of simple variable names, I believe it makes sense when it makes sense, and mathematical formulas are a great place: a and b is like shorthand for "items I'm comparing" or whatever. 👍 

I don't understand the formula itself, though. Or I understand it, but I don't think it has the result we want? Maybe you can explain it! I'll go through my thoughts here...

My first thought was that we should be comparing areas `(a.w * ah) - (b.w * b.h)`, but that won't have the appropriate result either, because it'll put items with same area but different dimensions (4x5 & 5x4) together, and it's the dimensions we need to match, not the area.

My thought now is that we don't need to sort, actually, and that you really CAN'T sort, because it's 2-dimensional and it's more about matching than ordering. What I think would be helpful would be to simply cluster identical items. So leave everything alone, except for 2 or more components that have the exact same w & h, then make sure they get processed sequentially so the later part of the algorithm keeps them on the same page.

Let me know what you think if you get a chance! I'll take a swipe at the algorithm after the kids go to bed and have something for you to look at before you become available.